mongodbatlas/mongodbatlas.go

```diff
@@ -260,8 +260,11 @@ func (c *Client) NewRequest(ctx context.Context, method, urlStr string, body int
 	if err != nil {
 		return nil, err
 	}
+	var buf io.Reader
+	if body != nil {
+		buf, err = c.newEncodedBody(body)
```

Comment:
you should move the check for error here
```suggestion
		if buf, err = c.newEncodedBody(body); err !=nil{
 			return nil, err
		}
```